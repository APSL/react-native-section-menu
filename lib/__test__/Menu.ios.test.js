import { View } from 'react-native'
import React from 'react'
import Menu from '../Menu.ios'
import renderer from 'react-test-renderer'

describe('Menu', () => {
  beforeEach(() => {
    jest.mock('TabBarIOS', () => {
      const RealComponent = require.requireActual('TabBarIOS')
      const React = require('React');
      class TabBarIOS extends React.Component {
        static Item = 'TabBarItemIOS'
        render() {
          return React.createElement('TabBarIOS', this.props, this.props.children);
        }
      }
      TabBarIOS.propTypes = RealComponent.propTypes;
      return TabBarIOS;
    });
  })
  test('Renders', () => {
    const tree = renderer.create(
      <Menu
        entries={[
          {id: 0, element: <View />, title: 'First'},
          {id: 1, element: <View />, title: 'Second'},
          {id: 2, element: <View />, title: 'Third'}
        ]}
      />
    )
    const component = tree.toJSON()
    expect(component).toMatchSnapshot()
  })
})
