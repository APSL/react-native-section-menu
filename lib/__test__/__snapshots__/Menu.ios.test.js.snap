exports[`Menu Renders 1`] = `
<TabBarIOS
  entries={
    Array [
      Object {
        "element": <class extends RealComponent {
          render() {
            return React.createElement(
            RealComponent.displayName,
            this.props,
            this.props.children);
    
          }} />,
        "id": 0,
        "title": "First"
      },
      Object {
        "element": <class extends RealComponent {
          render() {
            return React.createElement(
            RealComponent.displayName,
            this.props,
            this.props.children);
    
          }} />,
        "id": 1,
        "title": "Second"
      },
      Object {
        "element": <class extends RealComponent {
          render() {
            return React.createElement(
            RealComponent.displayName,
            this.props,
            this.props.children);
    
          }} />,
        "id": 2,
        "title": "Third"
      }
    ]
  }>
  <TabBarItemIOS
    element={
      <class extends RealComponent {
            render() {
              return React.createElement(
              RealComponent.displayName,
              this.props,
              this.props.children);
      
            }} />
    }
    id={0}
    onPress={[Function bound _onPressItem]}
    selected={true}
    title="First">
    <View />
  </TabBarItemIOS>
  <TabBarItemIOS
    element={
      <class extends RealComponent {
            render() {
              return React.createElement(
              RealComponent.displayName,
              this.props,
              this.props.children);
      
            }} />
    }
    id={1}
    onPress={[Function bound _onPressItem]}
    selected={false}
    title="Second">
    <View />
  </TabBarItemIOS>
  <TabBarItemIOS
    element={
      <class extends RealComponent {
            render() {
              return React.createElement(
              RealComponent.displayName,
              this.props,
              this.props.children);
      
            }} />
    }
    id={2}
    onPress={[Function bound _onPressItem]}
    selected={false}
    title="Third">
    <View />
  </TabBarItemIOS>
</TabBarIOS>
`;
